l2-contracts/src/L2RewardManager.sol

```diff
@@ -174,36 +197,55 @@ contract L2RewardManager is
     }
 
     /**
-     * @notice Reverts the already frozen interval. It bridges the xPufETH back to the L1
-     * @dev On the L1, we unwrap xPufETH to pufETH and burn the pufETH to undo the minting
+     * @notice Reverts the already frozen interval. It bridges the pufETH back to the L1
+     * @dev On the L1, we burn the pufETH to undo the minting
      * We use msg.value to pay for the relayer fee on the destination chain.
      */
-    function revertInterval(address bridge, uint256 startEpoch, uint256 endEpoch) external payable restricted {
-        _revertInterval(bridge, startEpoch, endEpoch);
+    function revertInterval(uint256 startEpoch, uint256 endEpoch) external payable restricted {
+        _revertInterval(startEpoch, endEpoch);
+    }
+
+    /**
+     * @notice Sets the delay period for claiming rewards
+     * @param delayPeriod The new delay period in seconds
+     */
+    function setDelayPeriod(uint256 delayPeriod) external restricted {
+        _setClaimingDelay(delayPeriod);
     }
 
     /**
-     * @notice Updates the bridge data.
-     * @param bridge The address of the bridge.
-     * @param bridgeData The updated bridge data.
+     * @notice Sets the address of the old pufETH token
+     * @dev If set to Zero address, means that the old pufETH token is not used anymore
+     * @param xPufETH The address of the old pufETH token
      */
-    function updateBridgeData(address bridge, BridgeData memory bridgeData) external restricted {
+    function setXPufETH(address xPufETH) external restricted {
         RewardManagerStorage storage $ = _getRewardManagerStorage();
+        $.xPufETH = xPufETH;
+    }
 
-        if (bridge == address(0)) {
+    /**
+     * @notice Sets the pufETH OFT address
+     * @param newPufETHOFT The new pufETH OFT address
+     */
+    function setPufETHOFT(address newPufETHOFT) external restricted {
+        if (newPufETHOFT == address(0)) {
             revert InvalidAddress();
         }
-
-        $.bridges[bridge].destinationDomainId = bridgeData.destinationDomainId;
-        emit BridgeDataUpdated(bridge, bridgeData);
+        RewardManagerStorage storage $ = _getRewardManagerStorage();
+        address oldPufETHOFT = $.pufETHOFT;
+        $.pufETHOFT = newPufETHOFT;
```

Comment:
can this be immutable as well?

I assume that we would have this deployed before upgrading this contract